docs/firmware/golioth-firmware-sdk/authentication/certificate-auth.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Certificate Authentication
+sidebar_position: 1
+---
+
+# Certificate Authentication with Golioth
+
+Golioth uses the X.509 standard with the Elliptic Curve Digital Signature
+Algorithm (ECDSA) for certificate authentication. This delivers much more
+robust security compared to Pre-Shared Key (PSK) authentication.
+
+Certificate authentication also eases the provisioning process. Devices do not
+need to be registered with Golioth prior to making the first connection. In
+manufacturing, units are granted individual device certificates signed using a
+trusted chain of root certificates and intermediate certificates. When the first
+connection is authenticated, Golioth will create a unique record of that device
+and it will appear as a member of your IoT fleet in your Golioth Project.
+
+## Overview
+
+You can test out Certificate Authentication with the following actions:
+
+* Create a root certificate
+* Create device certificates signed using the private root key
+* Upload the public root certificate to Golioth
+* Provide the device certificate and key to your device
+
+:::note Safeguard your private root key
+
+You retain the **private** root key, it is not shared with Golioth. Only you are
+able to sign new device certificates.
+
+:::
+
+## Generate Certificates
+
+### Create a self-signed root certificate
+
+For testing we will use `openssl` to generate a self-signed root certificate.
+
+```shell
+SERVER_NAME='golioth'
+
+# Generate an elliptic curve private key 
+# Run `openssl ecparam -list_curves` to list all available algorithms
+# Keep this key safe! Anyone who has it can sign authentic-looking device certificates
+openssl ecparam -name prime256v1 -genkey -noout -out "${SERVER_NAME}.key.pem"
+
+# Create and self-sign a corresponding public key / certificate
+openssl req -x509 -new -nodes \
+    -key "${SERVER_NAME}.key.pem" \
+    -sha256 -subj "/C=BR/CN=Root ${SERVER_NAME}" \
+    -days 1024 -out "${SERVER_NAME}.crt.pem"
+```
+
+This will generate the following files:
+
+* **Signed root certificate:** `golioth.crt.pem`
+* **Private root key:** `golioth.key.pem`
+
+### Create and sign device certificates
+
+Now generate and sign a device certificate. The resulting key/csr will be
```

Comment:
The CSR (certificate signing request) stays in PEM format. The signed certificate and the private key are in DER
```suggestion
Now generate and sign a device certificate. The resulting key/cert will be
```